### PR TITLE
Iteration5 al

### PR DESCRIPTION
**Pull request by alex-latham**
Adds tests/methods for worst_loss and head_to_head on TeamStats and StatTracker. Adds test space for seasonal_summary to TeamStats and StatTracker, and method to StatTracker.

Rspec passes all but one test (seasonal_summary) in 0.70948 seconds. Rake passes all tests except for two skipped seasonal_summary tests.

**Update 4:19 pm 2/26/20:**
Also includes refactor of GameStats class and two new Stats methods - `percentage`, and `average_hashes`. These two methods may be used in any of our subclasses. I'm very excited about `average_hashes`.

**Update 5:42 pm 2/26/20**
Also includes refactor of StatTracker to not expose games/teams/game_teams data array unnecessarily. As a consequence, `GameStats`, `LeagueStats`, `SeasonStats`, and `TeamStats` are refactored.

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -142,4 +137,13 @@ def worst_season(team_id)
...
+
+  def worst_loss(team_id)
+    @team_stats.worst_loss(team_id) end
+
+  def head_to_head(team_id)
+    @team_stats.head_to_head(team_id) end
+
+  def seasonal_summary(team_id)
+    @team_stats.seasonal_summary(team_id) end
 end
```
Comment (Maxwell-Baird) [inline lib/stat_tracker.rb:148]: good job having a holder here just to make sure we are in line count

File: test/team_stats_test.rb [hunk 2]
```diff
@@ -93,4 +92,96 @@ def test_count_wins
...
+         :average_goals_scored=>1.79,
+         :average_goals_against=>2.36},
+       :regular_season=>
+        {:win_percentage=>0.45,
+         :total_goals_scored=>178,
+         :total_goals_against=>159,
+         :average_goals_scored=>2.17,
+         :average_goals_against=>1.94}}}
+
+    assert_equal expected, @team_stats.seasonal_summary("3")
```
Comment (whitneykidd) [inline test/team_stats_test.rb:201]: This test is so visually pleasing

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -6,22 +6,22 @@
...
-    @league_stats = LeagueStats.new(@games, @teams, @game_teams)
-    @season_stats = SeasonStats.new(@games, @teams, @game_teams)
-    @team_stats = TeamStats.new(@games, @teams, @game_teams) end
+    games = csv_data(locations[:games], Game)
+    teams = csv_data(locations[:teams], Team)
+    game_teams = csv_data(locations[:game_teams], GameTeam)
+    @game_stats = GameStats.new(games)
+    @league_stats = LeagueStats.new(games, teams, game_teams)
+    @season_stats = SeasonStats.new(games, teams, game_teams)
+    @team_stats = TeamStats.new(games, teams, game_teams) end
```
Comment (tylertomlinson) [inline lib/stat_tracker.rb:24]: Makes sense to do it like this since we aren't using them anywhere else

**Review by tylertomlinson — APPROVED** [review]
Looks good. Alot of work done on this PR

**Pull request merged**